src/pages/profile.tsx

```diff
@@ -26,6 +42,18 @@ export default function Page() {
   const [newCity, setNewCity] = useState(user.city || '');
   const [newWechat, setNewWechat] = useState(user.wechat || '');
 
+  const { data: pref } = trpcNext.users.getUserPreference.useQuery({ userId: user.id });
+  const [unsaved, setUnsaved] = useState<UserPreference>();
+  useEffect(() => setUnsaved(pref), [pref]);
+
+  const updateInterviewPref = (data: InterviewPref) => {
+    if (data !== undefined) {
```

Comment:
why do you need this conditional? by type definition it can't be undefined.

Reply:
ah. you can do `invariant(data !== undefined)`. you need `import invariant from 'tiny-invariant'`